lib/storage/storage.go

```diff
@@ -2197,7 +2201,7 @@ func (s *Storage) add(rows []rawRow, dstMrs []*MetricRow, mrs []MetricRow, preci
 			pendingHourEntries = append(pendingHourEntries, genTSID.TSID.MetricID)
 		}
 
-		if logNewSeries {
+		if logNewSeries || (s.logNewSeriesEndTime.After(time.Now())) {
```

Comment:
Rename `logNewSeriesEndTime` to `logNewSeriesUntill`.

Reply:
I think  there is a race condition since `s.logNewSeriesEndTime` is modified in one gorouting and read in another. `s.logNewSeriesEndTime` has to be protected by an atomic.

Reply:
I have already fixed it.

Reply:
@makasim